java/org/contikios/cooja/Simulation.java

```diff
@@ -696,55 +748,39 @@ public boolean setConfigXML(Element root, boolean quick) throws Exception {
    *          Mote to remove
    */
   public void removeMote(final Mote mote) {
+    invokeSimulationThread(() -> doRemoveMote(mote));
+  }
 
-    /* Simulation is running, remove mote in simulation loop */
-    Runnable removeMote = new Runnable() {
-      @Override
-      public void run() {
-        motes.remove(mote);
-        currentRadioMedium.unregisterMote(mote, Simulation.this);
+  private void doRemoveMote(Mote mote) {
+    motes.remove(mote);
 
-        /* Dispose mote interface resources */
-        mote.removed();
-        for (MoteInterface i: mote.getInterfaces().getInterfaces()) {
-          i.removed();
-        }
-
-        setChanged();
-        notifyObservers(mote);
+    if (currentRadioMedium != null) {
```

Comment:
I don't think currentRadioMedium should be null at this stage, something is tearing down the radio medium earlier in the new code.